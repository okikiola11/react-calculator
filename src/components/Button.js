import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const { name, clickHandler } = props;

  return (
    <>
      <button onClick={() => clickHandler(name)} type="button" name={name}>
        {name}
      </button>
    </>
  );
};

Button.propTypes = {
  name: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
};

Button.defaultProps = {
  name: '',
};

export default Button;
