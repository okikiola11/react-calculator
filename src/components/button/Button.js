import React from 'react';
import PropTypes from 'prop-types';
import * as styles from './button.module.css';

const Button = (props) => {
  const {
    name, clickHandler, color, wide,
  } = props;

  const handleClick = (buttonName) => clickHandler(buttonName);

  return (
    <>
      <button
        onClick={() => handleClick(name)}
        type="button"
        className={`${color ? styles.grayColor : styles.orangeColor} ${wide && styles.wide ? styles.wide : styles.button}`}
      >
        {name}
      </button>
    </>
  );
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
  color: PropTypes.bool,
  wide: PropTypes.bool,
};

Button.defaultProps = {
  color: false,
  wide: false,
};

export default Button;
