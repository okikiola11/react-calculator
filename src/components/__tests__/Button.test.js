import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../button/Button';

describe('Button Component', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(<Button name="AC" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
