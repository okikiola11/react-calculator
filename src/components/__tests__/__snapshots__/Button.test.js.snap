// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<button
  className="undefined undefined"
  onClick={[Function]}
  type="button"
>
  AC
</button>
`;
